Tests/UnitTests/Core/Seeding/BinnedGroupTest.cpp

```diff
@@ -0,0 +1,256 @@
+  // We do not provide navigation, the constructor will define it
+  Acts::BinnedGroup<grid_3d_t> group_3d(std::move(grid_3d), *binFinder_3d.get(),
+                                        *binFinder_3d.get());
+
+  // Move Constructor/Assignment
+  const Acts::BinnedGroup<grid_1d_t> group_1d_moved(std::move(group_1d));
+  const Acts::BinnedGroup<grid_2d_t> group_2d_moved(std::move(group_2d));
+  const Acts::BinnedGroup<grid_3d_t> group_3d_moved = std::move(group_3d);
+
+  [[maybe_unused]] const grid_1d_t& retrievedGrid = group_1d.grid();
+}
+
+BOOST_AUTO_TEST_CASE(binned_group_iterations_1d_emptyGrid) {
+  using grid_t =
+      Acts::Grid<std::vector<std::size_t>, Acts::detail::EquidistantAxis>;
+  using binfinder_t = Acts::GridBinFinder<1ul>;
+
+  Acts::detail::EquidistantAxis xAxis(0, 100, 10);
+  grid_t grid(std::make_tuple(std::move(xAxis)));
+  std::unique_ptr<binfinder_t> binfinder = std::make_unique<binfinder_t>(0);
+  Acts::BinnedGroup<grid_t> group(std::move(grid), *binfinder.get(),
+                                  *binfinder.get());
+
+  Acts::BinnedGroupIterator<grid_t> itrStart = group.begin();
+  Acts::BinnedGroupIterator<grid_t> itrStop = group.end();
+
+  std::size_t nIterations = 0ul;
+  for (; itrStart != itrStop; ++itrStart, ++nIterations) {
+    [[maybe_unused]] auto candidates = *itrStart;
+  }
+
+  BOOST_CHECK_EQUAL(nIterations, 0ul);
+}
+
+BOOST_AUTO_TEST_CASE(binned_group_iterations_2d_emptyGrid) {
+  using grid_t =
+      Acts::Grid<std::vector<std::size_t>, Acts::detail::EquidistantAxis,
+                 Acts::detail::EquidistantAxis>;
+  using binfinder_t = Acts::GridBinFinder<2ul>;
+
+  Acts::detail::EquidistantAxis xAxis(0, 100, 10);
+  Acts::detail::EquidistantAxis yAxis(0, 100, 10);
+  grid_t grid(std::make_tuple(std::move(xAxis), std::move(yAxis)));
+  std::unique_ptr<binfinder_t> binfinder = std::make_unique<binfinder_t>(0, 0);
+  Acts::BinnedGroup<grid_t> group(std::move(grid), *binfinder.get(),
+                                  *binfinder.get());
+
+  Acts::BinnedGroupIterator<grid_t> itrStart = group.begin();
+  Acts::BinnedGroupIterator<grid_t> itrStop = group.end();
+
+  std::size_t nIterations = 0ul;
+  for (; itrStart != itrStop; ++itrStart, ++nIterations) {
+    [[maybe_unused]] auto candidates = *itrStart;
+  }
+
+  BOOST_CHECK_EQUAL(nIterations, 0ul);
+}
+
+BOOST_AUTO_TEST_CASE(binned_group_iterations_1d_perFilledGrid) {
+  using grid_t =
+      Acts::Grid<std::vector<std::size_t>, Acts::detail::EquidistantAxis>;
+  using binfinder_t = Acts::GridBinFinder<1ul>;
+
+  Acts::detail::EquidistantAxis xAxis(0, 100, 10);
+  grid_t grid(std::make_tuple(xAxis));
+  /// Add some entries to the grid
+  grid.at(1ul).push_back(4ul);
+  grid.at(1ul).push_back(1ul);
+  grid.at(8ul).push_back(7ul);
+  grid.at(9ul).push_back(2ul);
+
+  std::unique_ptr<binfinder_t> botBinfinder = std::make_unique<binfinder_t>(0);
+  std::unique_ptr<binfinder_t> topBinfinder = std::make_unique<binfinder_t>(1);
```

Comment:
Doesn't really matter for the test, but why `unique_ptr`?

Reply:
no idea, i forgot ... I'll remove the unique ptrs